### DISCUSSION TITLE
Extraction of trade routes ?

### DISCUSSION THREAD

**Discussion by JerjerBinks** [NiHoel/Anno1800UXEnhancer#16] [Ideas]
Hi !

Did you already try to extract trade routes ?
Or did I miss the possibility ?

And do you have a method to calculate how much T/min are traded in a route ? 
I find it pretty hard to estimate how much I need to put on a cargo because no time of travel is given.

**Comment by NiHoel** [comment]
No, I do not have time at the moment to figure that out. 

The only reliable way would be to extract the data from the running game.
There are two promising approaches:

## Python API
https://github.com/Nick11/anno1800_python_api_prober
(API documentation: https://github.com/suhrmann/Anno-1800-Python-API)

## DLL injection
https://github.com/egomeh/anno1800-scripting-api
